### DISCUSSION TITLE
Please teach me how to extract modal analysis results from an RST file.

### DISCUSSION THREAD

**Discussion by sokenMori** [ansys/pydpf-post#604] [Q&A] [answered]
After performing modal analysis, which function of dpf.post can be used to obtain the following values from the RST file?

Eigenvalues
Node coordinates

If this is my first time posting, please let me know if there are any mistakes.

**Comment by pmaroneh** [comment, marked as answer]
Hi @sokenMori , for the mesh, [this example](https://post.docs.pyansys.com/version/stable/examples/01-Detailed-Examples/05-mesh-exploration.html#sphx-glr-examples-01-detailed-examples-05-mesh-exploration-py) should help  and for extracting the frequencies, [this one](https://post.docs.pyansys.com/version/stable/examples/00-Different-analysis-types/02-modal-simulation.html#sphx-glr-examples-00-different-analysis-types-02-modal-simulation-py) should help. For these type of how-to questions, you can also use/search the [Ansys Developer Forum.](https://discuss.ansys.com/)

**Reply by sokenMori** [reply]
Thank you for telling me. It was very helpful, and I was able to obtain the eigenvalues.

Could you also tell me if it is possible to obtain the coordinates of the nodes in the same way? I have looked at the documentation, but I cannot find how to get them.

Thank you for the information about the Ansys Developer Forum. I will make use of it in the future.

**Reply by pmaroneh** [reply]
Hi @sokenMori , glad this helped for the eigen values. The first example I shared shows how to extract node coordinates (somewhere near the end of the example): 
![image](https://github.com/ansys/pydpf-post/assets/76440649/5a665ab1-52da-4916-9b28-7001a19f249a)

**Reply by sokenMori** [reply]
It was my lack of verification. I was able to obtain the coordinates. Thank you for showing me how.

It was my first post, but I am very grateful for your kind and courteous response.

**Reply by sokenMori** [reply]
I'm sorry for the inconvenience.

If you have time, could you please teach me a bit more? I was able to see the frequencies using simulation.time_freq_support, but I don't know how to obtain the values of those frequencies.

Also, I want to obtain the weight and center of gravity, but I can't because errors like the following occur with post.ModalMechanicalSimulation.elemental_mass() or centroids():

ElementalMass:2464<-Source operator "mapdl::rst::ElementalMass" not found
The requested Data format"fields_container" is not valid for this pin
If there is another method, please let me know.

**Reply by pmaroneh** [reply]
Thank you for your kind messages @sokenMori . Of course I can try to help further. I'd like to share with you one trick I use all the time. If you're unsure on how to get something from an object, check the different methods and properties available for this object by using `dir(object).` In your specific case:
![image](https://github.com/ansys/pydpf-post/assets/76440649/a4bf1b1a-62ba-474e-bf0e-0c728d106bfe)
returns (screenshot does not show the entire results, just the promising ones):
![image](https://github.com/ansys/pydpf-post/assets/76440649/14b3446b-03db-408c-9060-f9978b9d4655)
This is how I obtained:
![image](https://github.com/ansys/pydpf-post/assets/76440649/827f2bb3-571e-4fc2-90c4-f0521c307fe3)
So `simulation.time_freq_support.time_frequencies` is a field with one component. So to get the values you can use simulation.time_freq_support.time_frequencies.data which returns a DPF array and you can get the frequencies by iterating on the list:
![image](https://github.com/ansys/pydpf-post/assets/76440649/a066ad0a-c74a-46ec-a18a-5951b84dde4a)
or  
![image](https://github.com/ansys/pydpf-post/assets/76440649/5bbd9b89-f4e5-48e0-9b09-73f921e83e72)

**Reply by pmaroneh** [reply]
For the elemental_mass, I'll let @PProfizi chime in, as I'm getting an error similar as yours:
![image](https://github.com/ansys/pydpf-post/assets/76440649/3330012a-e8f8-4082-95e1-3015d4a423b4)
![image](https://github.com/ansys/pydpf-post/assets/76440649/bfbafd78-ac58-4b70-8845-eadcf0e57415)

**Reply by PProfizi** [reply]
Hi @sokenMori, hi @pmaroneh,
it seems the operator is missing indeed. I raised an issue server-side and will let you know.

**Reply by sokenMori** [reply]
Thank you for teaching me such an important technique, @pmaroneh. I was able to solve the problem related to frequency acquisition.

**Reply by PProfizi** [reply]
Hi @sokenMori @pmaroneh, as a workaround while the `ElementalMass` operator is unavailable, you can get the [elemental_volume](https://post.docs.pyansys.com/version/stable/api/ansys.dpf.post.modal_mechanical_simulation.html#ansys.dpf.post.modal_mechanical_simulation.ModalMechanicalSimulation.elemental_volume) result if available in your result file, and multiply it by the density.
If it is not available, you can also use `pydpf-core` to compute the elemental volume of the mesh with [this operator](https://dpf.docs.pyansys.com/version/stable/api/ansys.dpf.core.operators.geo.elements_volume.html).

**Reply by sokenMori** [reply]
Hi @PProfizi ,I'm sorry for the late reply. Thank you for the information. I will try to verify the process while referring to it.

**Reply by mcMunich** [reply]
@PProfizi  Is there a operator for elemental density? Because if you want to get an elemental mass, you need to know the density for a particular element. And I am not sure the material operator is working if we have to access a material from a elemental id.
